code/modules/interaction_tests/interaction_test.dm

```diff
@@ -0,0 +1,34 @@
+/datum/interaction_test
+	var/list/tracked_puppets
+	var/succeeded = TRUE
+	var/list/fail_reasons
+
+/datum/interaction_test/Destroy()
+	for(var/datum/test_puppeteer/puppet in tracked_puppets)
+		qdel(puppet)
+	return ..()
+
+/datum/interaction_test/proc/Run()
+	SHOULD_CALL_PARENT(FALSE)
+	Fail("Run() not implemented")
+
+/datum/interaction_test/proc/Fail(reason = "No reason")
+	succeeded = FALSE
+
+	if(!istext(reason))
+		reason = "FORMATTED: [reason != null ? reason : "NULL"]"
+
+	LAZYADD(fail_reasons, reason)
+
+/datum/interaction_test/proc/make_puppeteer(carbon_type = /mob/living/carbon/human)
+	var/datum/test_puppeteer/puppet = new(carbon_type)
```

Comment:
Might want a sanity check to verify that the passed type is indeed a carbon type here

Reply:
Done.